packages/nx-plugin/src/generators/configuration/code-pushup-config.ts

```diff
@@ -0,0 +1,110 @@
+import { Tree, generateFiles } from '@nx/devkit';
+import { join } from 'node:path';
+import { PersistConfig, UploadConfig } from '@code-pushup/models';
+import { ExtractArrays, ItemOrArray } from '@code-pushup/utils';
+
+export type ExecutableCode = {
+  fileImports: ItemOrArray<string>;
+  codeStrings: ItemOrArray<string>;
+};
+
+export type GenerateCodePushupConfigOptions = {
+  fileImports?: ItemOrArray<string>;
+  persist?: Partial<PersistConfig>;
+  upload?: Partial<UploadConfig>;
+  plugins?: ExecutableCode[];
+  categories?: ExecutableCode[];
+};
+
+export function generateCodePushupConfig(
```

Comment:
This is a more flexible generator function that will enable dynamic plugin registration for a test setup. For better understanding check out their tests as we do not use the full functionality ATM.

Reply:
Cc @Tlacenka